icechunk/src/session.rs

```diff
@@ -1516,7 +1598,17 @@ async fn flush(
 
     for (node_path, node_id) in flush_data.change_set.new_arrays() {
         trace!(path=%node_path, "New node, writing a manifest");
-        flush_data.write_manifest_for_new_node(node_id, node_path).await?;
+        // FIXME: grab the config
+        let config = ManifestShardingConfig::default();
```

Comment:
Need advice on how to thread the repo config down here

Reply:
The Session has the `RepositoryConfig`, I think it's OK to thread the sharding config through the few functions: do_commit, flush, etc. It's not great, those functions already have many args, but I think we can still stick a couple more before we refactor.